airflow/example_dags/plugins/workday.py

```diff
@@ -77,9 +77,10 @@ def next_dagrun_info(
                 tzinfo=UTC
             )
         else:  # This is the first ever run on the regular schedule.
-            next_start = restriction.earliest
-            if next_start is None:  # No start_date. Don't schedule.
+            if restriction.earliest is None:  # No start_date. Don't schedule.
                 return None
+            next_start = restriction.earliest
+
```

Comment:
This does not look relevant?

Reply:
Yeah, that is looks weird but `mypy` checks starts fail [here](https://github.com/apache/airflow/actions/runs/7253257238/job/19760229574)

```console
airflow/example_dags/plugins/workday.py:80: error: Incompatible types in
assignment (expression has type "Optional[DateTime]", variable has type
"DateTime")  [assignment]
                next_start = restriction.earliest
```

Reply:
Most likely due to Pendulum 3 has better annotation, the current code is indeed not type safe as Mypy points out, but the problem is masked by inferior annotations in Pendulum.

Reply:
See #36296